src/Web/Suavemente/Input.hs

```diff
@@ -152,3 +174,59 @@ enumDropdown
 enumDropdown label =
   dropdown label $ fmap (showMarkup &&& id) [minBound .. maxBound]
 
+------------------------------------------------------------------------------
+-- | A checkbox that turns off a class display
+checkboxShow :: String -> String -> Bool -> Suave Bool
+checkboxShow label cl =
+  mkInput parseJSON $ \name v ->
+  preEscapedString (showJs cl) <>
+  preEscapedString [qc|
+     <label for="{name}">{label}</label>
+     <input id="{name}" onchange="showJs('{cl}','{name}');onChangeFunc(event)" type="checkbox" {bool ("" :: String) "checked='checked'" v} autocomplete="off">
+     |]
+
+-- | js to show/hide a class based on a checkbox
+showJs :: String -> String
+showJs cl =
+  [qc|
+     <script>
+        function showJs (cl, box) \{
+          var vis = (document.getElementById(box).checked) ? "block" : "none";
+          Array.from(document.getElementsByClassName("{cl}")).forEach(x => x.style.display = vis);
+        };
+     </script>
+  |]
+
+-- | Modify the markup of a Suave
+markupF :: (Markup -> Markup) -> Suave a -> Suave a
+markupF f (Suave sa) = Suave $ do
+  Input markup s v <- sa
+  pure $ Input (f markup) s v
+
+-- | Wrap in a div
+div' :: String -> String -> Suave a -> Suave a
+div' cl st = markupF (\x ->
+  preEscapedString [qc|<div class="{cl}" style="{st}">|] <>
+  x <>
+  preEscapedString "</div>")
+
+-- | show/hide style string
+display :: Bool -> String
+display b = "display:" ++ bool "none" "block" b
+
+-- | apply a checkbox that turns a Suave a into a Suave (Maybe a)
+maybeInput :: String -> Bool -> Suave a -> Suave (Maybe a)
+maybeInput label start sa =
+  (\c a -> bool Nothing (Just a) c) <$>
+  checkboxShow label "wrap" start <*>
+  div' "wrap" (display start) sa
```

Comment:
you could just add this class to a table row, no?

Reply:
I dont think so, no.

Reply:
Maybe we can't get our hands on it as written today, but `tr`s definitely support classes and the `visibility` property